src/app.js

```diff
@@ -0,0 +1,124 @@
+const headline = (id, title, webUrl, apiUrl) => {
+  return {
+    id,
+    title,
+    webUrl,
+    apiUrl
+  }
+};
+
+class App {
+  constructor(makeHeadlinesRequest, makeSummaryRequest, makeArticleRequest) {
+    this.makeHeadlinesRequest = makeHeadlinesRequest;
+    this.makeSummaryRequest = makeSummaryRequest;
+    this.makeArticleRequest = makeArticleRequest;
+  }
+
+  getHeadlines () {
+    this.makeHeadlinesRequest(data => {
+      let idx = 0;
+      data.response.results.forEach((item) => {
+        idx++;
+        const newHeadline = headline (idx, item.webTitle, item.webUrl, item.apiUrl)
+        this._articleHeadline(newHeadline);
+      });
+    });
+  }
+
+  _getArticleImage = (articleDiv, articleUrl, idx) => {
+    this.makeArticleRequest(articleUrl, data => {
+      const doc = new DOMParser().parseFromString(data.response.content.fields.body, "text/html");
+      let imgUrl;
+      try {
+        imgUrl =  doc.querySelector("img").getAttribute("src");
+      } catch {
+        imgUrl = "./public/images/news.png";
+      }
+      this._articleImage(articleDiv, idx, imgUrl);
+    });
```

Comment:
Very impressive. I've not seen use of DOMParser before but the method looks good